test/with_dummyserver/test_https.py

```diff
@@ -74,31 +78,31 @@
 
 
 class TestHTTPS(HTTPSDummyServerTestCase):
-    tls_protocol_name = None
+    tls_protocol_name: Optional[str] = None
 
-    def tls_protocol_not_default(self):
+    def tls_protocol_not_default(self) -> bool:
         return self.tls_protocol_name in {"TLSv1", "TLSv1.1"}
 
-    def tls_version(self):
+    def tls_version(self) -> "ssl.TLSVersion":
         if self.tls_protocol_name is None:
             return pytest.skip("Skipping base test class")
         try:
             from ssl import TLSVersion
         except ImportError:
             return pytest.skip("ssl.TLSVersion isn't available")
-        return getattr(TLSVersion, self.tls_protocol_name.replace(".", "_"))
+        return getattr(TLSVersion, self.tls_protocol_name.replace(".", "_"))  # type: ignore[no-any-return]
```

Comment:
Enum types implement `__getitem__`  and the typing probably knows about it, so I think this can be

```py
return TLSVersion[self.tls_protocol_name.replace(".", "_")]
```